packages/@uppy/companion-client/src/Provider.js

```diff
@@ -111,7 +111,7 @@ export default class Provider extends RequestClient {
       const authWindow = window.open(link, '_blank')
       const handleToken = (e) => {
         if (e.source !== authWindow) {
-          this.uppy.log.warn('ignoring event from unknown source', e)
+          this.uppy.log(`ignoring event from unknown source ${JSON.stringify(e.data)}`, 'warning')
```

Comment:
maybe yea

Reply:
if `e.data` might not be valid, we should check that rather than rely on `try`/`catch`. But maybe it's always valid?

Reply:
Yea maybe it’s better that it crashes if the data is invalid? Then we can fix the bug when we know what to fix

Reply:
I dont think that’s going to help at all, because most things that are falsy will not cause JSON.stringify to fail afaik. What could make it fail however is circular references or BigInts anywhere in the tree

Reply:
Btw we should really improve our uppy logger so that it can take an arbitrary number of arguments, each either objects, errors or strings. Then we don’t have to manually do these things like json stringify when logging

Reply:
i added a try/catch just to be sure, because IMO a log statement should never throw an error